src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -431,21 +437,23 @@ const WorkflowView = _.flow(
       const methods = await Methods.list({ namespace: methodNamespace, name: methodName })
       const snapshotIds = _.map(m => _.pick('snapshotId', m).snapshotId, methods)
       const inputsOutputs = isRedacted ? {} : await Methods.configInputsOutputs(config)
+      const selection = workflowSelectionStore.get()
+      const modifiedConfig = selection.key === selectionKey ? _.set('rootEntityType', selection.entityType, config) : config
       this.setState({
-        savedConfig: config, modifiedConfig: config,
+        savedConfig: config, modifiedConfig,
         currentSnapRedacted: isRedacted, savedSnapRedacted: isRedacted,
         entityMetadata,
         savedInputsOutputs: inputsOutputs,
         modifiedInputsOutputs: inputsOutputs,
         snapshotIds,
         errors: isRedacted ? { inputs: {}, outputs: {} } : augmentErrors(validationResponse),
-        entitySelectionModel: this.resetSelectionModel(config.rootEntityType),
+        entitySelectionModel: this.resetSelectionModel(modifiedConfig.rootEntityType, selection.key === selectionKey ? selection.entities : {}),
```

Comment:
I need to think about this more. Was this a bug? If not, this looks like a change in intended behavior.

Reply:
Chatted in person. This was correct before since we were just setting `modifiedConfig` equal to `config`. Since they might be different now, we want to make sure to use `modifiedConfig`